src/posit/connect/vanities.py

```diff
@@ -0,0 +1,236 @@
+from typing import Callable, List, Optional, TypedDict
+
+from typing_extensions import NotRequired, Required, Unpack
+
+from .errors import ClientError
+from .resources import Resource, ResourceParameters, Resources
+
+
+class Vanity(Resource):
+    """A vanity resource.
+
+    Vanities maintain custom URL paths assigned to content.
+
+    Warnings
+    --------
+    Vanity paths may only contain alphanumeric characters, hyphens, underscores, and slashes.
+
+    Vanities cannot have children. For example, if the vanity path "/finance/" exists, the vanity path "/finance/budget/" cannot. But, if "/finance" does not exist, both "/finance/budget/" and "/finance/report" are allowed.
+
+    The following vanities are reserved by Connect:
+    - `/__`
+    - `/favicon.ico`
+    - `/connect`
+    - `/apps`
+    - `/users`
+    - `/groups`
+    - `/setpassword`
+    - `/user-completion`
+    - `/confirm`
+    - `/recent`
+    - `/reports`
+    - `/plots`
+    - `/unpublished`
+    - `/settings`
+    - `/metrics`
+    - `/tokens`
+    - `/help`
+    - `/login`
+    - `/welcome`
+    - `/register`
+    - `/resetpassword`
+    - `/content`
+    """
+
+    AfterDestroyCallback = Callable[[], None]
+
+    class VanityAttributes(TypedDict):
+        """Vanity attributes."""
+
+        path: Required[str]
+        content_guid: Required[str]
+        created_time: Required[str]
+
+    def __init__(
+        self,
+        /,
+        params: ResourceParameters,
+        *,
+        after_destroy: Optional[AfterDestroyCallback] = None,
+        **kwargs: Unpack[VanityAttributes],
+    ):
+        """Initialize a Vanity.
+
+        Parameters
+        ----------
+        params : ResourceParameters
+        after_destroy : AfterDestroyCallback, optional
+            Called after the Vanity is successfully destroyed, by default None
```

Comment:
This is interesting to me; what are its use cases?

Reply:
It's currently used on L145 below (`self._vanity._after_destroy = self.reset_vanity`), which says, "after the vanity is destroyed, call reset_vanity". It's a way to control the state of VanityMixin without having to interact with it directly from Vanity.